lib/web_ui/lib/src/engine/configuration.dart

```diff
@@ -55,14 +55,27 @@ FlutterConfiguration get configuration =>
   _configuration ??= FlutterConfiguration.legacy(_jsConfiguration);
 FlutterConfiguration? _configuration;
 
-/// Sets the given configuration as the current one.
+/// Overrides [configuration] with new values coming from `newConfig`.
+///
+/// If `newConfig` is null, the override is removed.
```

Comment:
When we remove the override in configuration, we should be going back to the default configuration values set in JS:

https://github.com/flutter/engine/blob/cd8cfa060e4e01f95101225129290f9fbb60f967/lib/web_ui/lib/src/engine/configuration.dart#L53-L56

The test that you proposed above should indeed pass, let's add it to the configuration tests!

Reply:
I added the test to the same test file that tests that the canvaskit variant is set by default (but that it can also be overridden).

I've also expanded the documentation of the method a little bit, but I'm not sure if it's doing it any favors for clarity or making it worse 😮‍💨, let me know!